client/state/happychat/middleware.js

```diff
@@ -18,6 +25,16 @@ export const requestTranscript = ( connection, { getState, dispatch } ) => {
 	);
 };
 
+export const sendRouteSetEventMessage = ( connection, { getState }, action ) =>{
+	const state = getState();
+	const currentUser = getCurrentUser( state );
+	if ( isHappychatClientConnected( state ) &&
+		isHappychatAcceptingChats( state ) &&
+		isHappychatChatAssigned( state ) ) {
+		connection.sendEvent( `Looking at https://wordpress.com${ action.path }?support_user=${ currentUser.username } ` );
```

Comment:
That's another option I've thought about. My concern with that is that it would make Calypso and Happychat more coupled, so any other client with the same needs would have to use Calypso Redux actions.

I acknowledge the power of passing the raw Redux actions, indeed! But at this point, I prefer making the translation between the Redux action and what's shown in the HUD in the customer client, so both things remain decoupled. This is definitely a thing worth exploring/revisiting for next iteration of this task, to learn how to pass more actions that are meaningful to HEs.